### PR TITLE
Amwal 241 update customer information handling in magento

### PR DESCRIPTION
**Pull request by ahmardi**

### CODE REVIEW

File: view/frontend/templates/express/checkout-button.phtml [hunk 1]
```diff
@@ -44,7 +45,7 @@ if ($customer->getDefaultShippingAddress()) {
         <?= $formSelector ? 'form-selector="' . $escaper->escapeHtmlAttr($formSelector) . '"' : '' ?>
          id="<?= $escaper->escapeHtmlAttr($buttonId) ?>" initial-email="<?= $escaper->escapeHtmlAttr($customer->getEmail()) ?>" initial-first-name="<?= $escaper->escapeHtmlAttr($customer->getFirstname()) ?>"
          initial-last-name="<?= $escaper->escapeHtmlAttr($customer->getLastname()) ?>" initial-phone-number="<?= $escaper->escapeHtmlAttr($phone_number ?? '') ?>"
-         initial-address="<?= $escaper->escapeHtmlAttr($formatedAddress ?? '') ?>">
+         initial-address="<?= $escaper->escapeHtmlAttr($formatedAddress ?? '') ?>" post-code-optional-countries="<?= $escaper->escapeHtmlAttr($postcodeOptionalCountries) ?>">
```
Comment (sameh-amwal) [inline view/frontend/templates/express/checkout-button.phtml:48]: I haven't looked closely here before, why do we keep add attributes to the containing div. the only two attributes needed are `class` and `trigger-context`

File: view/frontend/web/js/amwal-magento-react-button/src/AmwalMagentoReactButton.tsx [hunk 2]
```diff
@@ -275,6 +275,7 @@ const AmwalMagentoReactButton = ({
         enablePreCheckoutTrigger={config.enable_pre_checkout_trigger}
         onAmwalPreCheckoutTrigger={handleAmwalPreCheckoutTrigger}
         onUpdateOrderOnPaymentsuccess={handleUpdateOrderOnPaymentsuccess}
+        postcodeOptionalCountries={JSON.stringify(config.post_code_optional_countries) as any}
```
Comment (sameh-amwal) [inline view/frontend/web/js/amwal-magento-react-button/src/AmwalMagentoReactButton.tsx:278]:
```suggestion
        postcodeOptionalCountries={config.post_code_optional_countries}
```

File: ViewModel/ExpressCheckoutButton.php [hunk 3]
```diff
@@ -113,4 +123,20 @@ public function getFormSelector(?string $formSelector, ?ProductInterface $produc
...
+        $optionalCountries = [];
+        $getOptionalCountries = $this->storeConfig->getValue(
+            'general/country/optional_zip_countries',
+            ScopeInterface::SCOPE_STORE
+        );
+        if ($getOptionalCountries) {
+            $optionalCountries = explode(',', $getOptionalCountries);
+        }
+        return $optionalCountries;
+    }
```
Comment (sameh-amwal) [inline ViewModel/ExpressCheckoutButton.php:141]: is this function still needed?

Comment (ahmardi) [reply ViewModel/ExpressCheckoutButton.php:141]: Yes we use it to get the getOptionalCountries from the store settings

Comment (ahmardi) [reply ViewModel/ExpressCheckoutButton.php:141]: No, i will remove it

**Review by sameh-amwal — APPROVED** [review]

**Pull request merged**